packages/next/next-server/server/image-optimizer.ts

```diff
@@ -30,7 +30,13 @@ export async function imageOptimizer(
   parsedUrl: UrlWithParsedQuery
 ) {
   const { nextConfig, distDir } = server
-  const { sizes = [], domains = [] } = nextConfig?.images || {}
+  const { sizes = [], domains = [], loader } = nextConfig?.images || {}
+
+  if (loader !== 'default') {
+    await server.render404(req, res, parsedUrl)
```

Comment:
Instead of still triggering this logic should we not add the route at all [here](https://github.com/vercel/next.js/blob/6d3b065884ce84a355abb23f39b1d8f261bd87d0/packages/next/next-server/server/next-server.ts#L613-L619) if the default loader isn't used?

Reply:
@ijjk But we need to read the `nextConfig` object either way so I thought it made sense to keep it here so that its easy to understand

Reply:
Make sense 👍